modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RepositoryPlugin.java

```diff
@@ -86,7 +88,7 @@ protected S3Repository createRepository(
         final BigArrays bigArrays,
         final RecoverySettings recoverySettings
     ) {
-        return new S3Repository(metadata, registry, service.get(), clusterService, bigArrays, recoverySettings);
+        return new S3Repository(metadata, registry, service.get(), clusterService, bigArrays, recoverySettings, meter.get());
```

Comment:
Actual wiring seems fairly simple (`meter.set(telemetryProvider.getMeter());`).
Is there value setting in other repository plugins as well?

Reply:
Yeah it's simple enough. So I think it's also OK to leave other repositories out for now since adding it later is trivial. For now, I am mostly interested in getting s3 working.